kafka/producer_test.go

```diff
@@ -117,6 +129,22 @@ func TestNewProducerBasic(t *testing.T) {
 			//lint:ignore SA1012 passing a nil context is a valid use for this call.
 			fetches := client.PollRecords(nil, 1)
 			assert.Len(t, fetches.Records(), 0)
+
+			// Assert tracing happened properly
+			assert.Equal(t, len(exp.GetSpans()), spanCount+1)
```

Comment:
Shouldn't there also be a span from the `franz-go` library?

Reply:
Hmm probably. I'll look into that.
Also, the async test is somehow failing in CI, but not locally.

Reply:
Indeed, kotel wasn't enabled at all due to a configuration issue. I've fixed it.

Reply:
Thanks for looking into it.